src/iterable/zip.js

```diff
@@ -0,0 +1,35 @@
+import {map} from './map'
+
+/**
+ * unpacks an Iterable of Iterables and yields an array of items
+ *
+ * @this {Iterable<any>}
+ * @ntime O(n)
```

Comment:
This is a recurses two levels deep so it's `O(nm)` if I'm looking at it right. :P